ios/NebulaNetworkExtension/Site.swift

```diff
@@ -3,56 +3,124 @@ import MobileNebula
+            try container.encode(string)
+        case let .number(number):
+            try container.encode(number)
+        case let .bool(bool):
+            try container.encode(bool)
+        case .null:
+            try container.encodeNil()
+        }
     }
 
-    required init(coder aDecoder: NSCoder) {
-        id = aDecoder.decodeObject(forKey: "id") as! String
-        type = aDecoder.decodeObject(forKey: "type") as! String
-        message = aDecoder.decodeObject(forKey: "message") as Any?
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.singleValueContainer()
+
+        if let object = try? container.decode([String: JSON].self) {
+            self = .object(object)
+        } else if let array = try? container.decode([JSON].self) {
+            self = .array(array)
+        } else if let string = try? container.decode(String.self) {
+            self = .string(string)
+        } else if let bool = try? container.decode(Bool.self) {
+            self = .bool(bool)
+        } else if let number = try? container.decode(Double.self) {
+            self = .number(number)
+        } else if container.decodeNil() {
+            self = .null
+        } else {
+            throw DecodingError.dataCorrupted(
+                .init(codingPath: decoder.codingPath, debugDescription: "Invalid JSON value.")
+            )
+        }
     }
     
-    init(id: String, type: String, message: Any) {
-        self.id = id
-        self.type = type
-        self.message = message
+    func any() -> Any? {
+        switch self {
+        case let .array(array):
+            return array
+        case let .object(object):
+            return object
+        case let .string(string):
+            return string
+        case let .number(number):
+            return number
+        case let .bool(bool):
+            return bool
+        case .null:
+            return nil
+        }
     }
 }
 
-class IPCRequest: NSObject, NSCoding {
-    var type: String
-    var callbackId: String
-    var arguments: Dictionary<String, Any>?
-    
-    func encode(with aCoder: NSCoder) {
-        aCoder.encode(type, forKey: "type")
-        aCoder.encode(arguments, forKey: "arguments")
-        aCoder.encode(callbackId, forKey: "callbackId")
+
+extension JSON: CustomDebugStringConvertible {
+    public var debugDescription: String {
+        switch self {
+        case .string(let str):
+            return str.debugDescription
+        case .number(let num):
+            return num.debugDescription
+        case .bool(let bool):
+            return bool.description
+        case .null:
+            return "null"
+        default:
+            let encoder = JSONEncoder()
+            encoder.outputFormatting = [.prettyPrinted]
+            return try! String(data: encoder.encode(self), encoding: .utf8)!
+        }
     }
+}
+
+class IPCResponse: Codable {
+    var type: IPCResponseType
+    //TODO: change message to data?
```

Comment:
Is this just referring to the name of the var?

Reply:
Yes